File: Frontend/src/app/model/analyse-laboratoire.ts
import { StatutAnalyse } from './enums';
import { Patient } from './patient';
import { DossierMedical } from './dossier-medical';
import { ResultatAnalyse } from './resultat-analyse';

export interface AnalyseLaboratoire {
    id?: string;
    date?: string;
    type: string;
    fichierPDF?: string;
    statut?: StatutAnalyse;
    patient?: Patient;
    dossierMedical?: DossierMedical;
    resultats?: ResultatAnalyse[];
}

export interface AnalyseLaboratoireDTO {
    type: string;
    patientId: string;
}

File: backend/src/main/java/com/pfe/pfe/model/AnalyseLaboratoire.java
package com.pfe.pfe.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "analyses_laboratoire")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AnalyseLaboratoire {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false)
    private LocalDate date = LocalDate.now();
    
    @Column(nullable = false)
    private String type;
    
    private String fichierPDF;
    
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StatutAnalyse statut = StatutAnalyse.EN_ATTENTE;
    
    @ManyToOne
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;
    
    @ManyToOne
    @JoinColumn(name = "dossier_medical_id")
    private DossierMedical dossierMedical;
    
    @OneToMany(mappedBy = "analyse", cascade = CascadeType.ALL)
    private List<ResultatAnalyse> resultats = new ArrayList<>();
    
    public enum StatutAnalyse {
        EN_ATTENTE, EN_COURS, TERMINE, VALIDE
    }
}
